zerver/tests/test_email_notifications.py

```diff
@@ -626,6 +627,41 @@ def _extra_context_in_missed_stream_messages_mention_two_senders(
             msg_id, verify_body_include, email_subject, send_as_user, trigger="mentioned"
         )
 
+    def _resolved_topic_missed_stream_messages_thread_friendly(self, send_as_user: bool) -> None:
+        topic_name = "threading and so forth"
+        othello_user = self.example_user("othello")
+        msg_id = -1
+        for i in range(0, 3):
+            msg_id = self.send_stream_message(
+                othello_user,
+                "Denmark",
+                content=str(i),
+                topic_name=topic_name,
+            )
+
+        # This is based on a similar example in test_message_edit.py; there's a reasonable case to be made for
+        # pulling this into a shared wrapper function but for now, since this is only the second instance of
+        # needing to do this in tests and since the other uses client_patch, for now we'll leave the
+        # duplication.
+        resolution_result = self.api_patch(
+            othello_user,
+            "/json/messages/" + str(msg_id),
```

Comment:
Hmm, I wouldn't have expected `api_patch` to work when passed a `/json/` URL; in theory those URLs only support session authentication. The fact that this worked possibly represents a bug somewhere in either the testing system that we should investigate.

Reply:
It appears there's a decent bit of precedent for this (whether correct or erroneous), see this imperfect grep:

```
(vermillion) zulip  » resolved-topic-email-threading » rg 'api_post' | rg json
zerver/webhooks/bitbucket/tests.py:        result = self.api_post(self.test_user, self.url, payload, content_type="application/json,")
zerver/webhooks/bitbucket/tests.py:        result = self.api_post(self.test_user, self.url, payload, content_type="application/json,")
zerver/tests/test_event_system.py:            result = self.api_post(user, "/json/register")
zerver/tests/test_event_system.py:            result = self.api_post(user, "/json/register")
zerver/tests/test_event_system.py:            result = self.api_post(normal_user, "/json/register", dict(all_public_streams="true"))
zerver/tests/test_event_system.py:            result = self.api_post(guest_user, "/json/register", dict(all_public_streams="true"))
zerver/tests/test_user_groups.py:        result = self.api_post(sender, "/json/messages", payload)
zerver/tests/test_subs.py:        json_result = self.api_post(
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:            resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(iago, "/json/realm/playgrounds", payload)
zerver/tests/test_realm_playgrounds.py:        resp = self.api_post(hamlet, "/json/realm/playgrounds")
```

Looking at the `api_*` methods, they appear to jam a `HTTP_AUTHORIZATION` member into `extra` (eventually becoming HTTP Basic Auth via `encode_credentials`).

Given the existing uses, I'm inclined to think digging much further into the hows and whys of this working is a separate issue and separate PR from this; open to discussion there based on project norms I of course wouldn't be familiar with! If I'm understanding the existing URLs correctly, PATCH-ing `/api/v1/messages/{msg_id}` should accomplish the same thing here, so the abstraction of `ZulipTestCase#mark_topic_resolved` can still move forward here.

Thoughts?

Reply:
`ZulipTestCase#mark_topic_resolved` implemented in a54934f6f3 along with the URL change. Tests all still pass locally at this point.

Reply:
> It appears there's a decent bit of precedent for this

Yeah, I agree we should move this conversation off this PR. Do you want to just start a thread on #backend about this quirk? That's usually what we do with this sort of side detail where we're confused and discussion of it might want to spawn multiple issues -- one for doing the sweep and another for something around making this not work, but I'm not sure what the right plan is.

(I agree we should be happy to merge this PR without doing anything further on this point beyond making sure we don't lose track of this apparent tooling bug we noticed).

Reply:
Opened in CZO, resolving this thread

Reply:
This is 
https://chat.zulip.org/#narrow/stream/3-backend/topic/api_*.20methods.20in.20tests.20still.20pass.20against.20.2Fjson.20URLs/near/1458739